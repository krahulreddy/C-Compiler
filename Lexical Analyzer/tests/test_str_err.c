#include <stdio.h>

void main()
{
    int num = 2, i;
    char str_1 = "Hello";
    char str_2 = "World !!;

    printf("%s %s", str_1, str_2);
}