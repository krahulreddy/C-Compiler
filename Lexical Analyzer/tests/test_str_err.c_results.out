Line			Lexeme			Token Class
1		#include <stdio.h> 		Pre Processor directive

2
3			void 			KEYWORD
			main 			KEYWORD
			( 			OPENING BRACKETS
			) 			CLOSING BRACKETS

4			{ 			OPENING BRACES

5			int 			KEYWORD
			num 			IDENTIFIER
			= 			OPERATOR
			2 			NUMBER CONSTANT
			, 			COMMA DELIMITER
			i 			IDENTIFIER
			; 			SEMICOLON DELIMITER

6			char 			KEYWORD
			str_1 			IDENTIFIER
			= 			OPERATOR
			"Hello" 			STRING CONSTANT
			; 			SEMICOLON DELIMITER

7			char 			KEYWORD
			str_2 			IDENTIFIER
			= 			OPERATOR
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
ERROR: INCOMPLETE_STRING at line no. 7
	"
	^^
XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX


SYMBOL TABLE

	i			IDENTIFIER
	num			IDENTIFIER
	char			KEYWORD
	int			KEYWORD
	main			KEYWORD
	str_1			IDENTIFIER
	str_2			IDENTIFIER
	void			KEYWORD


CONSTANT TABLE

	"Hello"			STRING CONSTANT
	2			NUMBER CONSTANT
